bindings/pyroot/pythonizations/python/ROOT/_pythonization/_tmva/_batchgenerator.py

```diff
@@ -350,56 +353,78 @@ def ConvertBatchToPyTorch(self, batch: Any) -> torch.Tensor:
-        data.reshape((self.batch_size * self.num_columns,))
 
-        return_data = torch.Tensor(data).reshape(self.batch_size, self.num_columns)
+        if tuple(batch.GetShape()) != (self.batch_size,self.num_columns):
+            batch_size, num_columns = tuple(batch.GetShape())
+        else:
+            batch_size = self.batch_size
+            num_columns = self.num_columns
+
+        data.reshape((batch_size * num_columns,))
+
+        return_data = torch.Tensor(data).reshape(batch_size, num_columns)
 
         # Splice target column from the data if weight is given
         if self.target_given:
             target_data = return_data[:, self.target_index]
-            return_data = torch.column_stack(
-                (
-                    return_data[:, : self.target_index],
-                    return_data[:, self.target_index + 1 :],
-                )
-            )
+            return_data = torch.column_stack((return_data[:, : self.target_index], return_data[:, self.target_index + 1 :]))
 
             # Splice weights column from the data if weight is given
             if self.weights_given:
                 if self.target_index < self.weights_index:
                     self.weights_index -= 1
 
                 weights_data = return_data[:, self.weights_index]
-                return_data = torch.column_stack(
-                    (
-                        return_data[:, : self.weights_index],
-                        return_data[:, self.weights_index + 1 :],
-                    )
-                )
-                return return_data, target_data, weights_data
+                return_data = torch.column_stack((return_data[:, : self.weights_index], return_data[:, self.weights_index + 1 :]))
+
+                return return_data, target_data.reshape(-1,1), weights_data.reshape(-1,1)
 
-            return return_data, target_data
+            return return_data, target_data.reshape(-1,1)
 
         return return_data
 
     def ConvertBatchToTF(self, batch: Any) -> np.ndarray:
         """
-        PLACEHOLDER: at this moment this function only calls the
-        ConvertBatchToNumpy function. In the Future this function can be
-        used to convert to TF tensors directly
+        Convert a RTensor into a TensorFlow tensor
 
         Args:
             batch (RTensor): Batch returned from the RBatchGenerator
 
         Returns:
-            np.ndarray: converted batch
+            tensorflow.Tensor: converted batch
         """
-        # import tensorflow as tf
+        import tensorflow as tf
 
-        batch = self.ConvertBatchToNumpy(batch)
+        data = batch.GetData()
 
-        # TODO: improve this by returning tensorflow tensors
-        return batch
+        if tuple(batch.GetShape()) != (self.batch_size,self.num_columns):
+            batch_size, num_columns = tuple(batch.GetShape())
+        else:
+            batch_size = self.batch_size
+            num_columns = self.num_columns
```

Comment:
Similarly here, I fear that this may result in unexpected behaviour for the user in case they don't see the correct batch size returned to them